core/modules/system/system.install

```diff
@@ -3802,6 +3802,42 @@ function system_update_1102() {
   }
 }
 
+/**
+ * Create system.extensions config and populate with enabled modules and themes.
+ */
+function system_update_1103() {
+  $result = db_query("SELECT name, status, type, schema_version, info FROM {system} ORDER BY name ASC")->fetchAll(PDO::FETCH_ASSOC);
+  $modules = [];
+  $themes = [];
+  foreach ($result as $row) {
+    $extension = (array) $row;
+    $extension['info'] = unserialize($extension['info']);
+
+    if ($extension['type'] === 'module') {
+      // Do not include hidden modules in the list.
+      if (!empty($extension['info']['hidden'])) {
```

Comment:
We should test this with `ubercart` since it uses hidden modules in an unusual way. This would put ubercart sub-modules into the extensions file - like `cart` but would skip the mother module: `ubercart`.